R/data_modify.R

```diff
@@ -10,21 +10,29 @@
 #' - A sequence of named, literal expressions, where the left-hand side refers
 #'   to the name of the new variable, while the right-hand side represent the
 #'   values of the new variable. Example: `Sepal.Width = center(Sepal.Width)`.
-#' - A sequence of string values, representing expressions.
-#' - A variable that contains a string representation of the expression. Example:
+#' - A vector of length 1 (which will be recycled to match the number of rows
+#'   in the data), or of same length as the data.
+#' - A variable that contains a value to be used. Example:
 #'   ```r
-#'   a <- "2 * Sepal.Width"
-#'   data_modify(iris, a)
+#'   a <- "abc"
+#'   data_modify(iris, var_abc = a) # var_abc contains "abc"
+#'   ```
+#' - An expression can also be provided as string and wrapped in
+#' `as_expr()`. Example:
```

Comment:
It should be mentioned somewhere that `as_expr()` is not a "true" function in the sense that it helps us processing expressions in `data_modify()` but it cannot be used outside and is not exported / documented.

Reply:
We should then do the same for the select-helpers that are mentioned in the docs for the `select` argument.

Reply:
Ok but let's add it here for `as_expr()` since we're revamping the entire function anyway.